lib/server.js

```diff
@@ -80,6 +81,78 @@ class Server {
     });
   }
 
+  startSession(connection, application, strategy, credentials, callback) {
+    const createSession = (username) => {
```

Comment:
I'd extract it to a private function that takes `(connection, application, strategy, callback)` as arguments.